pkg/vectorize/momath/math.go

```diff
@@ -22,6 +22,22 @@ import (
 	"github.com/matrixorigin/matrixone/pkg/container/vector"
 )
 
+func Asin(arg, result *vector.Vector) error {
+	argCol := vector.MustTCols[float64](arg)
+	resCol := vector.MustTCols[float64](result)
+	nulls.Set(result.Nsp, arg.Nsp)
+	for i, v := range argCol {
+		if !nulls.Contains(arg.Nsp, (uint64)(i)) {
+			if v < -1 || v > 1 {
+				nulls.Add(result.Nsp, uint64(i))
```

Comment:
This should error, not generating nulls.

Reply:
Now `asin` or `acos` behavior is:

1. if value is not a number, it will raise a convert error before executing the `math.Asin` function
2. if value is `NULL`, it will return `NULL`
3. if value is not in $[-1, 1]$, it will return `NULL`
4. if value is in $[-1, 1]$, it will return a number

And [mysql 8.0](https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_asin) behavior is:

1. if value is not a number, it will return $0$
2. if value is `NULL`, it will return `NULL`
3. if value is not in $[-1, 1]$, it will return `NULL`
4. if value is in $[-1, 1]$, it will return a number

So should we raise an error in case 3?

Reply:
Both 1 and 3 should error.